### PR TITLE
Remove old syntax from table and elem in parser

### PR DESCRIPTION
**Pull request by martianboy**
We've been keeping old syntax in the text format parser although they've
been removed from the parser and hardly any test case relies on them.
This PR will remove old syntax support for tables and element segments
and simplify the corresponding parser functions. A few test files were
affected by this that are updated.

### CODE REVIEW

**Comment by martianboy** [comment]
@tlively @aheejin @kripken Reviews appreciated here, too. (Alpine build step seems to be stuck.)

File: src/wasm/wasm-s-parser.cpp [hunk 1]
```diff
@@ -3209,13 +3202,37 @@ void SExpressionWasmBuilder::parseTable(Element& s, bool preParseImport) {
...
   }
-  if (i == s.size()) {
-    wasm.addTable(std::move(table));
-    return;
+
+  bool hasExplicitLimit = false;
+
+  if (s[i]->isStr()) {
+    char c = s[i]->c_str()[0];
+    if ('0' <= c && c <= '9') {
```
Comment (tlively) [inline src/wasm/wasm-s-parser.cpp:3210]: You can use `wasm::isInteger` to check for numbers here and below.

**Review by tlively — APPROVED** [review]

File: src/wasm/wasm-s-parser.cpp (hunk 1 above)
Comment (martianboy) [reply src/wasm/wasm-s-parser.cpp:3210]: I'm not sure if I understand the point here, `wasm::isInteger` seems to check that a `double` is actually an integer and also not infinite.

Comment (tlively) [reply src/wasm/wasm-s-parser.cpp:3217]: Sorry, I misremembered the function name. I meant `String::isNumber` :)

Comment (martianboy) [reply src/wasm/wasm-s-parser.cpp]: Oh nice! Thanks. Fixed now. By the way, do we parse hex numbers in general? Is there a utility for detecting and parsing hex the way it's defined and used in the spec? e.g. `0x1000_0000`

Comment (tlively) [reply src/wasm/wasm-s-parser.cpp]: No, not that I know of.

**Pull request merged**